ocaml/typing/typedecl.ml

```diff
@@ -528,6 +531,79 @@ let verify_unboxed_attr unboxed_attr sdecl =
       end
   end
 
+(* Note [Default layouts in transl_declaration]
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   For every type declaration we create in transl_declaration, we must
+   choose the layout to use in the [type_layout] field.
+
+   1. If there is a layout annotation, use that. We might later compute
+      a more precise layout for the type (e.g. [type t : value = int] or
+      [type t : value = A | B | C]); this will be updated in
+      [update_decl_layout], which also ensures that the updated layout is
+      a sublayout of the annotated layout.
+
+   2. If there is no annotation but there is a manifest, use the layout
+      of the manifest.
```

Comment:
Note that it's updated later in `check_coherence`?

(There's an additional subtlety this comment doesn't get into which is that any of the layout calculations mentioned may be inaccurate because the environment used in transl declaration doesn't have manifests for mutually defined types.  That's true about case 2 here, but is also true about the other cases.  It's right not to try to explain both things at once, but maybe include a reference to the comment in `enter_type` somewhere in here).